internal/frrsockets/frrsockets.go

```diff
@@ -54,12 +58,22 @@ func executeCmd(socketPath, cmd string, timeout time.Duration) ([]byte, error) {
 		return nil, err
 	}
 
-	// frr vty sockets expect command to be null-terminated
+	buf := make([]byte, 4096)
+
+	// Mimic vtysh by switching to 'enable' mode first. Note that commands need to be
+	// null-terminated.
+	if _, err = conn.Write([]byte("enable\x00")); err != nil {
+		return nil, err
+	}
+	if _, err := conn.Read(buf); err != nil {
```

Comment:
Does the socket return anything meaningful that can validate entering enable mode?

Reply:
`strace` simply shows it return four null chars... or perhaps null-terminated sequence of three null chars ;-)
```
write(3, "enable\0", 7)                 = 7 <0.000006>
recvmsg(3, {msg_name=NULL, msg_namelen=0, msg_iov=[{iov_base="\0\0\0\0", iov_len=4095}], msg_iovlen=1, msg_controllen=0, msg_flags=0}, 0) = 4 <0.000013>
```
If switching to enable mode fails, I would expect the subsequent command to fail, and we catch errors suitably on that one.

Reply:
Cool. Makes sense